### ISSUE TITLE
Updating extension results in AttributeError (null derefence)

### ISSUE THREAD

**Issue by oerpli** [Azure/azure-cli#9174]
**Describe the bug**
I tried to update an extension, which resulted in an AttributeError

**To Reproduce**
` az extension update --name azure-cli-iot-ext` results in:

```
The command failed with an unexpected error. Here is the traceback:

'NoneType' object has no attribute 'get'
Traceback (most recent call last):
  File "...\knack\knack\cli.py", line 206, in invoke
  File "...\azure-cli-core\azure\cli\core\commands\__init__.py", line 351, in execute
  File "...\azure-cli-core\azure\cli\core\commands\__init__.py", line 409, in _run_jobs_serially
  File "...\azure-cli-core\azure\cli\core\commands\__init__.py", line 402, in _run_job
  File "...\six\six.py", line 693, in reraise
  File "...\azure-cli-core\azure\cli\core\commands\__init__.py", line 379, in _run_job
  File "...\azure-cli-core\azure\cli\core\commands\__init__.py", line 171, in __call__
  File "...\azure-cli-core\azure\cli\core\__init__.py", line 451, in default_command_handler
  File "...\azure-cli-core\azure\cli\core\extension\operations.py", line 258, in update_extension
  File "...\azure-cli-core\azure\cli\core\extension\__init__.py", line 115, in get_version
AttributeError: 'NoneType' object has no attribute 'get'
```

**Expected behavior**
Updating extensions should not result in exceptions. 

**Environment summary**
Windows 10 1709, installed with MSI, version 2.0.61 (not updated  to 2.0.62 as I have read that it is broken).

**Additional context**
Issue encountered on corporate network with horrible network security settings. If this error is a result from that, I think at least the error message/Exception should reflect that in some way.

**Comment by oerpli** [comment]
Possibly related to #6441

**Comment by tjprescott** [comment]
@oerpli what version of the IOT extension do you currently have?  Use `az --version`

**Comment by oerpli** [comment]
0.3.7

```
>az --version
azure-cli                         2.0.61 *

acr                                2.2.3 *
acs                               2.3.20 *
advisor                            2.0.0
ams                                0.4.3 *
appservice                        0.2.16 *
backup                             1.2.2 *
batch                              4.0.0
batchai                            0.4.8
billing                            0.2.1
botservice                         0.1.9 *
cdn                                0.2.1 *
cloud                              2.1.1
cognitiveservices                  0.2.5
command-modules-nspkg               2.0.2
configure                         2.0.20 *
consumption                        0.4.2
container                         0.3.15 *
core                              2.0.61 *
cosmosdb                           0.2.9 *
dla                                0.2.5
dls                                0.1.8 *
dms                                0.1.3
eventgrid                          0.2.2 *
eventhubs                          0.3.4
extension                          0.2.4 *
feedback                           2.1.4 *
find                               0.3.1 *
hdinsight                          0.3.2 *
interactive                        0.4.2 *
iot                                0.3.7 *
iotcentral                         0.1.6
keyvault                          2.2.13 *
kusto                              0.2.1 *
lab                                0.1.6 *
maps                               0.3.4
monitor                           0.2.11 *
network                            2.3.5 *
nspkg                              3.0.3
policyinsights                     0.1.2
profile                            2.1.4 *
rdbms                              0.3.9 *
redis                              0.4.2
relay                              0.1.4
reservations                       0.4.2
resource                          2.1.12 *
role                               2.4.3 *
search                             0.1.1
security                           0.1.1
servicebus                         0.3.4
servicefabric                     0.1.15 *
signalr                            1.0.0
sql                                2.2.0 *
sqlvm                              0.1.1
storage                            2.3.2 *
telemetry                          1.0.2
vm                                2.2.17 *
```

**Comment by oerpli** [comment]
As I've just noticed that 2.0.63 has been released a few hours ago I updated to that version.

Running `az extension update --name azure-cli-iot-ext` results in: 
>No updates available for 'azure-cli-iot-ext'. Use --debug for more information.
 which seems fine. So either the issue only arises when there actually is an update that can be performed or the bug has been fixed somewhere between 2.0.61 and 2.0.63. 

As far as I am concerned this issue can be closed - I may reopen it if I encounter the bug again.
